pkg/devspace/configure/image.go

```diff
@@ -136,65 +137,73 @@ func (m *manager) AddImage(imageName, image, dockerfile, contextPath string, doc
 		m.log.WriteString("\n")
 		m.log.Info("DevSpace does *not* require pushing your images to a registry but let's assume you wanted to do that (optional)")
 
+		registryOptions := []string{skipRegistry, useDockerHub, useGithubRegistry, useOtherRegistry}
+		isLoggedIntoRegistry := false
 		authConfig, err := dockerClient.GetAuthConfig(dockerHubHostname, true)
 		if err == nil && authConfig.Username != "" {
 			useDockerHub = useDockerHub + fmt.Sprintf(registryUsernameHint, authConfig.Username)
 			registryDefaultOption = useDockerHub
+			registryOptions = []string{useDockerHub, useGithubRegistry, useOtherRegistry, skipRegistry}
+			isLoggedIntoRegistry = true
 		}
 
-		authConfig, err = dockerClient.GetAuthConfig(githubContainerRegistry, true)
-		if err == nil && authConfig.Username != "" {
-			useGithubRegistry = useGithubRegistry + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+		if !isLoggedIntoRegistry {
+			authConfig, err = dockerClient.GetAuthConfig(githubContainerRegistry, true)
+			if err == nil && authConfig.Username != "" {
+				useGithubRegistry = useGithubRegistry + fmt.Sprintf(registryUsernameHint, authConfig.Username)
+				registryDefaultOption = useGithubRegistry
+				registryOptions = []string{useGithubRegistry, useDockerHub, useOtherRegistry, skipRegistry}
+				isLoggedIntoRegistry = true
+			}
 		}
 
-		registryOptions := []string{useDockerHub, useGithubRegistry, useOtherRegistry}
 		selectedRegistry, err := m.log.Question(&survey.QuestionOptions{
-			Question:     "Which registry would you want to use to push images to? (optional, choose any)",
+			Question:     "Which registry would you want to use to push images to?",
 			DefaultValue: registryDefaultOption,
 			Options:      registryOptions,
 		})
 		if err != nil {
 			return err
 		}
 
-		registryHostname := ""
+		if selectedRegistry != skipRegistry {
+			registryHostname := ""
+			// if selectedRegistry == useDockerHub do nothing.
```

Comment:
This seems not correct to me as now the logic would ask for a registry hostname if you select the docker hub option.

Reply:
Oh right.